src/VisualStudio/Core/Def/StackTraceExplorer/StackFrameViewModel.cs

```diff
@@ -0,0 +1,310 @@
+            var argEndIndex = _frame.ArgsSpan.End + 1;
+
+            var methodText = _frame.OriginalText[_frame.MethodSpan.Start..argEndIndex];
+            var methodLink = new Hyperlink();
+            var methodClassifiedText = new ClassifiedText(ClassificationTypeNames.MethodName, methodText);
+            methodLink.Inlines.Add(MakeClassifiedRun(ClassificationTypeNames.MethodName, methodText));
+            methodLink.Click += (s, a) => NavigateToSymbol();
+            methodLink.RequestNavigate += (s, a) => NavigateToSymbol();
+            yield return methodLink;
+
+            if (_frame is ParsedFrameWithFile frameWithFile)
+            {
+                var textBetweenLength = frameWithFile.FileSpan.Start - argEndIndex;
+                var textBetweenSpan = new TextSpan(argEndIndex, textBetweenLength);
+                if (textBetweenSpan.Length > 0)
+                {
+                    var textBetween = _frame.OriginalText.Substring(textBetweenSpan.Start, textBetweenSpan.Length);
+                    yield return new Run(textBetween);
+                }
+
+                var fileText = _frame.OriginalText[frameWithFile.FileSpan.Start..frameWithFile.FileSpan.End];
+                var fileHyperlink = new Hyperlink();
+                fileHyperlink.Inlines.Add(MakeClassifiedRun(ClassificationTypeNames.Text, fileText));
+                fileHyperlink.RequestNavigate += (s, e) => NavigateToFile();
+                fileHyperlink.Click += (s, e) => NavigateToFile();
+                yield return fileHyperlink;
+
+                var end = frameWithFile.FileSpan.End;
+                if (end < _frame.OriginalText.Length)
+                {
+                    yield return MakeClassifiedRun(ClassificationTypeNames.Text, _frame.OriginalText[..end]);
+                }
+            }
+            else
+            {
+                if (argEndIndex < _frame.OriginalText.Length)
+                {
+                    yield return MakeClassifiedRun(ClassificationTypeNames.Text, _frame.OriginalText[argEndIndex..]);
+                }
+            }
+        }
+
+        private Document? GetDocument(out int lineNumber)
+        {
+            if (_cachedDocument is not null)
+            {
+                lineNumber = _cachedLineNumber;
+                return _cachedDocument;
+            }
+
+            var potentialMatches = GetFileMatches(out _cachedLineNumber);
+            if (potentialMatches.Any())
+            {
+                _cachedDocument = potentialMatches.First();
+            }
+
+            lineNumber = _cachedLineNumber;
+            return _cachedDocument;
+        }
+
+        private async Task<ISymbol?> GetSymbolAsync(CancellationToken cancellationToken)
+        {
+            if (_cachedSymbol is not null)
+            {
+                return _cachedSymbol;
+            }
+
+            _cachedSymbol = await _frame.ResolveSymbolAsync(_workspace.CurrentSolution, cancellationToken).ConfigureAwait(false);
```

Comment:
This is just a general comment, not specific to this line, but the balance of code in models, viewmodels and frame models feels a little weird to me. There is quite a broad hierarchy in the frame classes, but also logic for resolving symbols, but then in this class further up there is logic that is based on knowing what type of frame is being displayed. Just feels a little out of balance.

Reply:
The intended breakdown is hierarchy should be minimal of what to display, and then if it supports navigation or not. Navigation is needed for list double click handling here, otherwise it could purely be hyperlink based. That said, I'll take another look at this holistically and see if I can simplify. This definitely changed a lot as I progressed and it's possible it's just a mishmash of ideas.

Reply:
I moved all the parsing to `GetXXX` methods on the parsed items. Hopefully that clears things up.